integration-tests/chaos/ocr2vrf_chaos_test.go

```diff
@@ -0,0 +1,276 @@
+	}{
+		// network-* and pods-* are split intentionally into 2 parallel groups
+		// we can't use chaos.NewNetworkPartition and chaos.NewFailPods in parallel
+		// because of jsii runtime bug, see Makefile
+		PodChaosFailMinorityNodes: {
+			ethereum.New(defaultOCR2VRFEthereumSettings),
+			chainlink.New(0, defaultOCR2VRFSettings),
+			chaos.NewFailPods,
+			&chaos.Props{
+				LabelsSelector: &map[string]*string{ChaosGroupMinority: a.Str("1")},
+				DurationStr:    "1m",
+			},
+		},
+		//todo - currently failing, need to investigate deeper
+		//PodChaosFailMajorityNodes: {
+		//	ethereum.New(defaultOCR2VRFEthereumSettings),
+		//	chainlink.New(0, defaultOCR2VRFSettings),
+		//	chaos.NewFailPods,
+		//	&chaos.Props{
+		//		LabelsSelector: &map[string]*string{ChaosGroupMajority: a.Str("1")},
+		//		DurationStr:    "1m",
+		//	},
+		//},
+		//todo - do we need these chaos tests?
+		//PodChaosFailMajorityDB: {
+		//	ethereum.New(defaultOCR2VRFEthereumSettings),
+		//	chainlink.New(0, defaultOCR2VRFSettings),
+		//	chaos.NewFailPods,
+		//	&chaos.Props{
+		//		LabelsSelector: &map[string]*string{ChaosGroupMajority: a.Str("1")},
+		//		DurationStr:    "1m",
+		//		ContainerNames: &[]*string{a.Str("chainlink-db")},
+		//	},
+		//},
+		//NetworkChaosFailMajorityNetwork: {
+		//	ethereum.New(defaultOCR2VRFEthereumSettings),
+		//	chainlink.New(0, defaultOCR2VRFSettings),
+		//	chaos.NewNetworkPartition,
+		//	&chaos.Props{
+		//		FromLabels:  &map[string]*string{ChaosGroupMajority: a.Str("1")},
+		//		ToLabels:    &map[string]*string{ChaosGroupMinority: a.Str("1")},
+		//		DurationStr: "1m",
+		//	},
+		//},
+		//NetworkChaosFailBlockchainNode: {
+		//	ethereum.New(defaultOCR2VRFEthereumSettings),
+		//	chainlink.New(0, defaultOCR2VRFSettings),
+		//	chaos.NewNetworkPartition,
+		//	&chaos.Props{
+		//		FromLabels:  &map[string]*string{"app": a.Str("geth")},
+		//		ToLabels:    &map[string]*string{ChaosGroupMajority: a.Str("1")},
+		//		DurationStr: "1m",
+		//	},
+		//},
+	}
+
+	for testcaseName, testCase := range testCases {
+		t.Run(testcaseName, func(t *testing.T) {
+			t.Parallel()
+			testNetwork := networks.SelectedNetwork
+			testEnvironment := environment.
+				New(&environment.Config{
+					NamespacePrefix: fmt.Sprintf(
+						"chaos-ocr2vrf-%s",
+						strings.ReplaceAll(strings.ToLower(testNetwork.Name),
+							" ",
+							"-")),
+					Test: t,
+				}).
+				AddHelm(testCase.networkChart).
+				AddHelm(testCase.clChart)
+			err := testEnvironment.Run()
+			require.NoError(t, err, "Error running test environment")
+
+			//todo - 3 nodes instead of 2 are down when specifying 1-2
+			err = testEnvironment.Client.LabelChaosGroup(testEnvironment.Cfg.Namespace, 1, 1, ChaosGroupMinority)
+			require.NoError(t, err)
+			err = testEnvironment.Client.LabelChaosGroup(testEnvironment.Cfg.Namespace, 3, 5, ChaosGroupMajority)
+			require.NoError(t, err)
+
+			chainClient, err := blockchain.NewEVMClient(testNetwork, testEnvironment)
```

Comment:
I like to keep tests not too abstract and no too detailed, so that you would read the code as list of steps. I didnt want to abstract away these steps